tests/MongoDB.Driver.Examples/InsertDataWithEncryptedFieldsWithAutoEncryption.cs

```diff
@@ -0,0 +1,295 @@
+            var medicalRecordsNamespace = CollectionNamespace.FromFullName("medicalRecords.patients");
+
+            var base64KeyId = ""; // paste the generated key in Standard GuidRepresentation form here
+
+            var ssnQuery = Builders<BsonDocument>.Filter.Eq("ssn", __sampleSsnValue);
+
+            // Construct a JSON Schema
+            var schema = JsonSchemaCreator.CreateJsonSchema(base64KeyId);
+
+            // Construct an encrypted client
+            var encryptedClient = CreateEncryptedClient(
+                medicalRecordsNamespace,
+                keyVaultNamespace,
+                schema);
+            var collection = encryptedClient
+                .GetDatabase(medicalRecordsNamespace.DatabaseNamespace.DatabaseName)
+                .GetCollection<BsonDocument>(medicalRecordsNamespace.CollectionName);
+
+            // Insert a document into the collection
+            collection.UpdateOne(ssnQuery, new BsonDocument("$set", SampleDoc), new UpdateOptions() { IsUpsert = true });
+            Console.WriteLine("Successfully upserted the sample document!");
+
+            // Query SSN field with encrypted client
+            var result = collection.Find(ssnQuery).Single();
+
+            Console.WriteLine("Encrypted client query by the SSN (deterministically-encrypted) field:\n" + result.ToJson());
+
+            // Query SSN field with normal client without encryption
+            var normalMongoClient = new MongoClient(__connectionString);
+            collection = normalMongoClient
+              .GetDatabase(medicalRecordsNamespace.DatabaseNamespace.DatabaseName)
+              .GetCollection<BsonDocument>(medicalRecordsNamespace.CollectionName);
+            var normalClientResult = collection.Find(ssnQuery).FirstOrDefault();
+            if (normalClientResult != null)
+            {
+                throw new Exception("Assert that the filtered data has not been found.");
+            }
+
+            // Query name (non-encrypted) field with normal client without encryption
+            var nameQuery = Builders<BsonDocument>.Filter.Eq("name", __sampleNameValue);
+            var normalClientNameResult = collection.Find(nameQuery).FirstOrDefault();
+            if (normalClientNameResult == null)
+            {
+                throw new Exception("Assert that the filtered data has been found.");
+            }
+
+            Console.WriteLine($"Query by name returned the following document:\n {normalClientNameResult}.");
+        }
+
+        private IMongoClient CreateEncryptedClient(
+            CollectionNamespace medicalRecordsNamespace,
+            CollectionNamespace keyVaultNamespace,
+            BsonDocument schema)
+        {
+            var kmsProviders = new Dictionary<string, IReadOnlyDictionary<string, object>>();
+
+            // For local master key
+            var localMasterKey = File.ReadAllText("master-key.txt");
```

Comment:
All other variables that contain a base64 value have "base64" in the name:

    var localMasterKeyBase64 =  ...